apps/src/netsim/Packet.js

```diff
@@ -224,16 +295,31 @@ Packet.Encoder.prototype.getBodyAsAscii = function (binary, bitsPerChar) {
  */
 Packet.Encoder.prototype.makeBinaryHeaders = function (headers) {
   var binaryHeaders = {};
-  this.formatSpec_.forEach(function (headerField){
-    if (headers.hasOwnProperty(headerField.key)) {
-      binaryHeaders[headerField.key] = dataConverters.intToBinary(
-          headers[headerField.key], headerField.bits);
+  this.headerSpec_.forEach(function (headerField){
+    if (headers.hasOwnProperty(headerField)) {
```

Comment:
Hmm; I'm inclined to leave this one, just in a gut "how it reads" sense.  It's less of a precondition check and more of an explicit logic step, saying "For each possible header, if that header is in the provided object, convert it."